src/python/sortingview/gui/extensions/mountainview/MVSortingView/PreloadCheck.tsx

```diff
@@ -17,16 +17,16 @@ type Props = {
 }
 
 const PreloadCheck: FunctionComponent<Props> = ({ recording, sorting, children, width, height, snippetLen }) => {
-    const sortingObject = sorting.sortingObject
+    const sortingObject = sorting ? sorting.sortingObject : undefined
     const recordingObject = recording.recordingObject
-    const runningState = useRef<{sortingObject: any, recordingObject: any}>({sortingObject: sorting.sortingObject, recordingObject: recording.recordingObject})
+    const runningState = useRef<{sortingObject: any, recordingObject: any}>({sortingObject, recordingObject})
 
     const matchesRunningState = useMemo(() => ((x: {recordingObject: any, sortingObject: any}) => (
         (runningState.current.sortingObject === x.sortingObject) && (runningState.current.recordingObject === x.recordingObject)
     )), [])
 
     const {channelName} = useChannel()
-    const {task: preloadExtractSnippetsTask} = usePureCalculationTask('preload_extract_snippets.2', {recording_object: recordingObject, sorting_object: sortingObject, snipets_len: snippetLen}, {channelName})
+    const {task: preloadExtractSnippetsTask} = usePureCalculationTask((recordingObject && sortingObject) ? 'preload_extract_snippets.2' : '', {recording_object: recordingObject, sorting_object: sortingObject, snipets_len: snippetLen}, {channelName})
```

Comment:
I take it passing an empty string will result in basically a no-op from `usePureCalculationTask`?

Reply:
Yes... it would return task=undefined. It would be better if `undefined` were used instead of empty string. I'll make an issue for that.

Reply:
https://github.com/magland/sortingview/issues/58